### PR TITLE
append EIP only to sslip.io dns_hosts

### PR DESCRIPTION
**Pull request by bacongobbler**
This enables users to use custom domain names like `fish.ca` or `fermyon.dev` without the technical requirements of a reverse DNS server like sslip.io.

This enables users to set a wildcard A record pointing to the elastic IP address. For example, here's my current setup in Cloudflare for https://hippo.f1sh.ca:

image

### CODE REVIEW

File: quick-start/terraform/main.tf [hunk 1]
```diff
@@ -100,7 +100,7 @@ resource "aws_instance" "ec2" {
 
   user_data = templatefile("${path.module}/scripts/user-data.sh",
     {
-      dns_zone                = "${aws_eip.lb.public_ip}.${var.dns_host}",
+      dns_zone                = aws_eip.lb.public_ip == "sslip.io" ? "${aws_eip.lb.public_ip}.${var.dns_host}" : "${var.dns_host}",
```
Comment (vdice) [inline quick-start/terraform/main.tf:103]:
```suggestion
      dns_zone                = var.dns_host == "sslip.io" ? "${aws_eip.lb.public_ip}.${var.dns_host}" : "${var.dns_host}",
```

File: quick-start/terraform/outputs.tf [hunk 2]
```diff
@@ -21,12 +21,12 @@ output "dns_host" {
 
 output "bindle_url" {
   description = "The URL for the Bindle server"
-  value       = "https://bindle.${aws_eip.lb.public_ip}.${var.dns_host}/v1"
+  value       = aws_eip.lb.public_ip == "sslip.io" ? "https://bindle.${aws_eip.lb.public_ip}.${var.dns_host}/v1" : "https://bindle.${var.dns_host}/v1"
```
Comment (vdice) [inline quick-start/terraform/outputs.tf:24]:
```suggestion
  value       = var.dns_host == "sslip.io" ? "https://bindle.${aws_eip.lb.public_ip}.${var.dns_host}/v1" : "https://bindle.${var.dns_host}/v1"
```

File: quick-start/terraform/outputs.tf [hunk 3]
```diff
@@ -21,12 +21,12 @@ output "dns_host" {
...
 }
 
 output "hippo_url" {
   description = "The URL for the Hippo server"
-  value       = "https://hippo.${aws_eip.lb.public_ip}.${var.dns_host}"
+  value       = aws_eip.lb.public_ip == "sslip.io" ? "https://hippo.${aws_eip.lb.public_ip}.${var.dns_host}" : "https://hippo.${var.dns_host}"
```
Comment (vdice) [inline quick-start/terraform/outputs.tf:29]:
```suggestion
  value       = var.dns_host == "sslip.io" ? "https://hippo.${aws_eip.lb.public_ip}.${var.dns_host}" : "https://hippo.${var.dns_host}"
```

**Review by vdice — COMMENTED** [review]
Nice!  Excited to test.  Meanwhile, just a few fixups on which value to check for 'sslip.io'...

**Comment by bacongobbler** [comment]
> * Do we want to add an example line showing how to override the dns_host var with a user-supplied value?  Somewhere in https://github.com/fermyon/nomad-aws-demo/tree/main/quick-start#how-to-deploy ?

Good call. Yes. I will go add that to this PR.

**Comment by bacongobbler** [comment]
Good point about the Let's Encrypt timing issue. I agree - let's feel this one out with user feedback. Perhaps we could add a note in the documentation describing that issue. That could help users wondering why their domain certificate is not being issued by LE.

**Review by vdice — APPROVED** [review]
Perfetto!  LGTM.

**Pull request merged**